src/Bab/RabbitMq/Action/DryRunAction.php

```diff
@@ -0,0 +1,163 @@
+<?php
+namespace Bab\RabbitMq\Action;
+
+use Bab\RabbitMq\HttpClientInterface;
+use Bab\RabbitMq\Response;
+use Bab\RabbitMq\Action\Formatter\Log;
+
+class DryRunAction extends Action
+{
+    const LABEL_EXCHANGE = 'exchange';
+    const LABEL_QUEUE = 'queue';
+    const LABEL_BINDING = 'binding';
+    const LABEL_PERMISSION = 'permission';
+
+    private $log;
+
+    public function __construct(HttpClientInterface $httpClient)
+    {
+        parent::__construct($httpClient);
+
+        $this->log = new Log();
+
+        $this->httpClient->enableDryRun(true);
+    }
+
+    public function endMapping()
+    {
+        $this->log->setLogger($this->logger);
+        $this->log->output();
+    }
+
+    public function resetVhost()
+    {
+        $vhost = $this->getContextValue('vhost');
+        $user = $this->getContextValue('user');
+
+        $this->log(sprintf('Will Delete vhost: <info>%s</info>', $vhost));
+
+        $this->log(sprintf('Will Create vhost: <info>%s</info>', $vhost));
+
+        $this->log(sprintf(
+            'Will Grant all permission for <info>%s</info> on vhost <info>%s</info>',
+            $user,
+            $vhost
+        ));
+    }
+
+    public function createExchange($name, $parameters)
+    {
+        $this->compare('/api/exchanges/'.$this->getContextValue('vhost').'/'.$name, $name, $parameters, self::LABEL_EXCHANGE);
+
+        return;
+    }
+
+    public function createQueue($name, $parameters)
+    {
+        $this->compare('/api/queues/'.$this->getContextValue('vhost').'/'.$name, $name, $parameters, self::LABEL_QUEUE);
+
+        return;
+    }
+
+    public function createBinding($name, $queue, $routingKey, array $arguments = array())
+    {
+        $vhost = $this->getContextValue('vhost');
+        $response = $this->query('GET', '/api/queues/'.$vhost.'/'.$queue.'/bindings');
+
+        $binding = array(
+            'source' => $name,
+            'destination' => $queue,
+            'vhost' => $vhost,
+            'routing_key' => is_null($routingKey) ? '' : $routingKey,
+            'arguments' => $arguments
+        );
+
+        if ($response->isSuccessful()) {
+            $bindings = json_decode($response->body, true);
+            foreach ($bindings as $existingBinding) {
+                $configurationDelta = $this->array_diff_assoc_recursive($binding, $existingBinding);
+
+                if (empty($configurationDelta)) {
+                    $this->log->addUnchanged(self::LABEL_BINDING, $queue.':'.$name, $arguments);
+                    return;
+                }
+            }
+        }
+
+        $this->log->addUpdate(self::LABEL_BINDING, $queue.':'.$name, $arguments);
```

Comment:
Could it be more interesting to log `exchange:routingKey`? (with the queue name also if you want. Maybe something like `exchange:routingKey -> queue`)
Otherwise it's possible to have multiple times the same output. (when multiple bindings with different routing_keys front an exchange to a queue).